### DISCUSSION TITLE
Movement by subwords (snake_case, camelCase, ...)

### DISCUSSION THREAD

**Discussion by JeftavanderHorst** [helix-editor/helix#5463] [Ideas] [answered]
Helix currently supports moving by both words and WORDS using the `w`, `b` and `e` keys. It might be interesting to add a third kind of word movement: subwords. This would consider the elements of snake_case and camelCase (and perhaps other casings) as separate words.

Example of what I mean (`|` characters indicate selection):

Moving by word:

- `|f|oo bar baz` -> press `w` -> `|foo |bar baz`
- `|f|oo_bar_baz` -> press `w` -> `|foo_bar_baz|`
- `|f|ooBarBaz` -> press `w` -> `|fooBarBaz|`

Moving by subword:

- `|f|oo bar baz` -> press `w` -> `|foo |bar baz`
- `|f|oo_bar_baz` -> press `w` -> `|foo_|bar_baz`
- `|f|ooBarBaz` -> press `w` -> `|foo|BarBaz`

I'm not sure what appropriate keybindings would be, since both `wbe` and `WBE` are already in use. For me personally, I never use WORDS so I would probably remap those keys to subwords, but others might disagree there. In any case, adding the commands (`move_next_subword_start`, etc) would be useful even if they don't have keybinds by default.

An example of similar behavior in vim is [vim-wordmotion](https://github.com/chaoren/vim-wordmotion).

If people are interested, I might have time to work on a PR for this.

**Comment by themissingcow** [comment]
I'd find this really valuable - it's one of the reasons I still use neovim at work. I don't know Rust but, I'd love to help work on this.

**Comment by JeftavanderHorst** [comment, marked as answer]
For anyone stumbling upon this discussion, I opened a PR for this here: https://github.com/helix-editor/helix/pull/8147
